### ISSUE TITLE
预测的结果和原句子一样

### ISSUE THREAD

**Issue by Amber921463001** [microsoft/NeuralSpeech#18]
I not only use 4 GPU, but also set update-freq=4. So if you just finetune FastCorrect on one card, you need to set update-freq=16.
From my log you can know all the hyper-parameters, you can have a check with yours.

_Originally posted by @YichongLeng in https://github.com/microsoft/NeuralSpeech/issues/14#issuecomment-1086579972_

我已经按照您设定的参数微调了模型，但我预测的结果和原句子一样，请问您知道这是什么原因吗？
我预测的输出是translated = [transf_gec.decode(hypos[0]['tokens']) for hypos in batched_hypos][0]

**Comment by Amber921463001** [comment]
我刚发现预测的结果不是和原句子一模一样，而是预测效果不好，大部分的预测结果和原句子一样，识别不到错误

**Comment by YichongLeng** [comment]
Since the WER of ASR model is less than 5%, 95% tokens are correct and do not need modification. You can test the WER before and after the correction model. You can also calculate the WER reduction (WERR) of the finetuned model and check it with WERR in our paper.

**Comment by Amber921463001** [comment]
我明白，但是微调后的模型对于测试集中的错误部分几乎识别不出来

**Comment by Amber921463001** [comment]
我看了您之前发的微调日志， train | epoch 031 | loss 2.111 | nll_loss 0.29 | wer_dur_loss 0.004 | word_ins 2.068 | to_be_edited_loss 0.083 | ppl 4.32 | wps 28333 | ups 0.2 | wpb 140692 | bsz 8575.8 | num_updates 882 | lr 0.00011025 | gnorm 0.231 | loss_scale 128 | train_wall 129 | wall 0，这个是您最终的损失吗?

**Comment by YichongLeng** [comment]
> 我明白，但是微调后的模型对于测试集中的错误部分几乎识别不出来

Could you please use more accurate description such as "model can only detect xx% of the error in test set" instead of “几乎”. In my result, our model can correct about 15% of error in the test set.



> 我看了您之前发的微调日志， train | epoch 031 | loss 2.111 | nll_loss 0.29 | wer_dur_loss 0.004 | word_ins 2.068 | to_be_edited_loss 0.083 | ppl 4.32 | wps 28333 | ups 0.2 | wpb 140692 | bsz 8575.8 | num_updates 882 | lr 0.00011025 | gnorm 0.231 | loss_scale 128 | train_wall 129 | wall 0，这个是您最终的损失吗?

Yes, I only finetune for 20 epoch (from epoch 10 to epoch 30) and select checkpoint by the dev set performance of AISHELL-1.

**Comment by Amber921463001** [comment]
我用自己的代码计算了一下测试集的wer，纠错前wer=4.83，纠错后wer=4.87，没有成功复现您的结果。我还想用您的脚本文件再计算一下wer，请问如果要运行cal_wer_aishell.sh，我应该把模型预测结果的文件放在什么位置呢？预测结果的文件应该是什么格式？

**Comment by Amber921463001** [comment]
我计算了一下测试集的wer，纠错前wer=4.83，纠错后wer=4.87，没有复现您的结果，请问我应该在哪方面去优化呢?

**Comment by YichongLeng** [comment]
It's strange. Maybe you can check the log difference between yours and mine since the log contains all **hyper-parameters** and **loss**. If the problem persists, maybe you can share your finetuning scripts and the result "data.json" from your finetuned model to me. I will have a check when I am free.

**Comment by Amber921463001** [comment]
我又重新调了参数，这次成功复现了您的结果，非常感谢您耐心的回复！

**Comment by dahu1** [comment]
> 我又重新调了参数，这次成功复现了您的结果，非常感谢您耐心的回复！

请问层主 是怎么调整参数的？我也遇到你这个问题了。

**Comment by wnntju** [comment]
> 
请问在哪里可以找到作者的训练log，参数如何确定。默认参数似乎不好
